### DISCUSSION TITLE
DIFERENÇA DE DADOS ENTRE TABELAS

### DISCUSSION THREAD

**Discussion by jornalistaCarlosOsorio** [sqlitebrowser/sqlitebrowser#2894] [General]
Tenho duas tabelas, mas apenas parcela dos dados estão sendo cruzados. Por exemplo: existem duas tabelas com 542 itens cada. Contudo, apenas 536 itens estão fazendo correlação. Tem alguma chave para que eu possa saber quais são os dados que não estão sendo cruzados?

**Comment by chrisjlocke** [comment]
Translation:
I have two tables, but only portion of the data is being crossed. For example, there are two tables with 542 items each. However, only 536 items are correlated. Do you have any key so I can know what data is not being crossed?

Apologies for this translation - just pasting what Google gave me!

Você tem um exemplo dos dados?
Você pode unir as duas tabelas e, em seguida, visualizar os campos que são nulos.
Portanto, se você tiver a tabela A com três linhas: Alice, Brian e Charlie.
E a tabela B, que tem três linhas, Brian, David e Eric.
Se juntássemos essas tabelas em nome, teríamos
Alice | nulo
Brian | Brian
Charlie | nulo

(existem diferentes junções - esta é uma junção à esquerda)

Podemos ver que Brian é compatível, mas Alice e Charlie não são compatíveis. 


Translation:

Do you have an example of the data?
You can join the two tables, then view the fields which are null.
So if you have table A which has three rows: Alice, Brian and Charlie.
And table B which has three rows, Brian, David and Eric.
If we joined these tables together on name, we would have
Alice | null
Brian | Brian
Charlie | null

(there are different joins - this is a left join)

We can see Brian is matched, but Alice and Charlie are not matched.

**Comment by jornalistaCarlosOsorio** [comment]
SELECT * FROM tabela_01 LEFT JOIN tabela_2
ON a.dados_tabela_1 = b.dados_tabela_2

**Reply by chrisjlocke** [reply]
Thats correct.  Decide which table is the 'main' table you want to view all records for, and which table you will link to.  
(Está correto. Decida qual tabela é a tabela 'principal' para a qual você deseja ver todos os registros e a qual tabela você fará o link.)

```
select * from main_table
left join other_table on main_table.field=other_table.field
```

The field you join on must be the same type - text, number, etc.
(O campo no qual você se junta deve ser do mesmo tipo - texto, número, etc.)

If you want to view just the records that match, add
(Se você quiser ver apenas os registros que correspondem, adicione)
`where other_table.field is not null`

If you want to view just the records that don't match, add
(Se você quiser ver apenas os registros que não correspondem, adicione)
`where other_table.field is not null`

**Comment by jornalistaCarlosOsorio** [comment]
Meu caro, tudo bem? Você pode me ajudar nesta duvida? O assunto é  **Juntar várias colunas em uma linha #3117** 

Muito obrigado,

**Reply by chrisjlocke** [reply]
Multiple columns from different tables?  Multiple columns from one row?

If you can explain how the database is laid out and what you need, we can assist. :)

--

Várias colunas de tabelas diferentes? Várias colunas de uma linha?

Se você puder explicar como o banco de dados é organizado e o que você precisa, podemos ajudar. :)
